utils/sender/src/main.rs

```diff
@@ -126,20 +130,162 @@ fn generate_receivers(n: usize, sender_index: usize) -> Vec<sp_core::crypto::Acc
+/// Note that signing is a CPU bound task, and hence this cannot be async.
+/// As a consequence, we use spawn_blocking here and communicate with the main thread using an unbounded channel.
+fn parallel_signing(
+	api: &Api,
+	threads: &usize,
+	n_tx_sender: usize,
+	producer: tokio::sync::mpsc::UnboundedSender<
+		Vec<
+			Result<
+				SubmittableExtrinsic<PolkadotConfig, subxt::OnlineClient<PolkadotConfig>>,
+				subxt::Error,
+			>,
+		>,
+	>,
+) -> Result<(), Error> {
+	let ext_deposit_addr = runtime::constants().balances().existential_deposit();
+	let genesis_hash = api.genesis_hash();
+	let ext_deposit = api.constants().at(&ext_deposit_addr)?;
+
+	for i in 0..*threads {
+		let api = api.clone();
+		let producer = producer.clone();
+		tokio::task::spawn_blocking(move || {
+			debug!("Thread {}: preparing {} transactions", i, n_tx_sender);
+			let ext_deposit = ext_deposit.clone();
+			let genesis_hash = genesis_hash.clone();
+			let receivers = generate_receivers(n_tx_sender, i);
+			let mut txs = Vec::new();
+			for j in 0..n_tx_sender {
+				debug!("Thread {}: preparing transaction {}", i, j);
+				let shift = i * n_tx_sender;
+				let signer = generate_signer(shift + j);
+				debug!("Thread {}: generated signer {}{}", i, DERIVATION, shift + j);
+				let tx_params = Params::new().era(Era::Immortal, genesis_hash);
+				let tx_payload = runtime::tx()
+					.balances()
+					.transfer_keep_alive(receivers[j as usize].clone().into(), ext_deposit);
+				let signed_tx =
+					match api.tx().create_signed_with_nonce(&tx_payload, &signer, 0, tx_params) {
+						Ok(signed) => Ok(signed),
+						Err(e) => Err(e),
+					};
+				txs.push(signed_tx);
+			}
+			match producer.send(txs) {
+				Ok(_) => (),
+				Err(e) => error!("Thread {}: failed to send transactions to consumer: {}", i, e),
+			}
+			info!("Thread {}: prepared and signed {} transactions", i, n_tx_sender);
+		});
+	}
+	Ok(())
+}
+
+/// Here, the signed extrinsics are submitted.
+/// TODO: Optimise this
+async fn submit_txs(
+	consumer: &mut tokio::sync::mpsc::UnboundedReceiver<
+		Vec<
+			Result<
+				SubmittableExtrinsic<PolkadotConfig, subxt::OnlineClient<PolkadotConfig>>,
+				subxt::Error,
+			>,
+		>,
+	>,
+	chunk_size: usize,
+	threads: usize,
+) -> Result<(), Error> {
+	let mut maybe_submittables = Vec::new();
+	while let Some(signed_txs) = consumer.recv().await {
+		debug!("Consumer: received {} submittable transactions", signed_txs.len());
+		maybe_submittables.push(signed_txs);
+		if threads == maybe_submittables.len() {
+			debug!("Consumer: received all submittable transactions, now starting submission");
+			let mut hashes = Vec::new();
+			for v in &maybe_submittables {
+				for chunk in v.chunks(chunk_size) {
+					for signed_tx in chunk {
+						match signed_tx {
+							Ok(tx) => {
+								let hash = tx.submit();
+								hashes.push(hash);
+							},
+							Err(e) => panic!("Error: {:?}", e),
+						}
+					}
+				}
+			}
+			try_join_all(hashes).await?;
```

Comment:
I think this will trigger all the `submit` task at once, we want to send in chunks right?

Reply:
Yes, you are right actually, good spot.